src/components/image/index.jsx

```diff
@@ -2,35 +2,161 @@ import PropTypes from "prop-types";
 
 const COMPONENT_CLASS_NAME = "c-image";
 
-const Image = ({ alt, className, height, loading, src, width }) => (
-	<img
-		alt={alt}
-		className={className ? `${COMPONENT_CLASS_NAME} ${className}` : `${COMPONENT_CLASS_NAME}`}
-		height={height}
-		loading={loading}
-		src={src}
-		width={width}
-	/>
-);
+const Image = ({
+	alt,
+	className,
+	loading,
+	src,
+	resizedOptions,
+	resizerURL,
+	responsiveImages,
+	sizes,
+}) => {
+	// get the default height and width of the image
+	// use aspect ratio to generate other sizes
+	const { width, height, auth } = resizedOptions;
+
+	if (!auth) {
+		// eslint-disable-next-line no-console
+		console.error("No auth token provided for resizer");
+
+		return (
+			<img
+				alt={alt}
+				className={className ? `${COMPONENT_CLASS_NAME} ${className}` : COMPONENT_CLASS_NAME}
+				src={src}
+				width={width}
+				height={height}
+			/>
+		);
+	}
+
+	// add all options except height and width
+	const stringOptionsWithoutHeightWidth = Object.keys(resizedOptions).reduce((acc, key) => {
+		if (key === "height" || key === "width") {
+			return acc;
+		}
+
+		if (acc === "") {
+			return `?${key}=${resizedOptions[key]}`;
+		}
+
+		return `${acc}&${key}=${resizedOptions[key]}`;
+	}, "");
+
+	// "https://resizer.com" + "\image.jpg" + "?auth=secret&filter=true"
+	const srcWithOptionsWithoutHeightWidth = resizerURL.concat(src, stringOptionsWithoutHeightWidth);
+
+	let aspectRatio = 0;
+
+	// if no height and width, no responsive image calculation
+	if (height && width) {
+		aspectRatio = width / height;
+	}
+
+	const responsiveHeightsAndWidths = responsiveImages.reduce(
+		(accumulator, responsiveImageWidth) => {
+			if (Number.isInteger(responsiveImageWidth) && responsiveImageWidth > 0) {
+				return [
+					...accumulator,
+					{
+						width: responsiveImageWidth,
+						// divide the derived aspect ratio by each of the responsiveImages widths to get the height
+						// aspect ratio of zero will not show the height
+						...(aspectRatio !== 0 && { height: responsiveImageWidth / aspectRatio }),
+					},
+				];
+			}
+			return accumulator;
+		},
+		[]
+	);
+
+	// add the height and width to the default src if they exist
+	// auth token will at least exist here so don't need to worry about ? prepend
+	const defaultSrc = srcWithOptionsWithoutHeightWidth.concat(
+		`${width ? `&width=${width}` : ""}${height ? `&height=${height}` : ""}`
+	);
+
+	return (
+		<img
+			alt={alt}
+			className={className ? `${COMPONENT_CLASS_NAME} ${className}` : COMPONENT_CLASS_NAME}
+			height={height}
+			loading={loading}
+			src={defaultSrc}
+			width={width}
+			srcSet={
+				responsiveHeightsAndWidths.length > 0
+					? responsiveHeightsAndWidths
+							.map(
+								(responsiveImage) =>
+									`${srcWithOptionsWithoutHeightWidth}&width=${responsiveImage.width}${
+										responsiveImage?.height ? `&height=${responsiveImage.height}` : ""
+									} ${responsiveImage.width}w`
+							)
+							.join(", ")
+					: null
```

Comment:
this can be written to not have to check the length:
```suggestion
				responsiveHeightsAndWidths
						.map(
							(responsiveImage) =>
								`${objectToQueryParams(responsiveImage)} ${responsiveImage.width}w`
						)
						.join(", ") || null
```
* not needing to check the length since a 0 length is a falsy string
* not needing the `srcWithOptionsWithoutHeightWidth` since removeFalsy removes falsy url variables ... may want to check for null or undefined instead since false, 0, or blank COULD be valid

With support from:
```
function objectToQueryParams(obj) {
	new URLSearchParams(removeFalsy(obj)).toString();
}
function removeFalsy(obj) {
	return Object.fromEntries(Object.entries(obj).filter(([_, v]) => v));
}
```

Reply:
Curious though, is there a default for `width`?

Reply:
default for responsive images array is empty arr. for responsiveImage.width, the user has total control over that?

Reply:
updated

Reply:
I updated to remove the length check and use short-circuiting based on the feedback. I'm not sure I understand the objectToQueryParams function: we need the resizer, the image name itself, and the search params. 

The current `srcWithOptionsWithoutHeightWidth` has this now. I'm using search params method to construct that string already.

Reply:
Doesn't really work with what you have now, but I have some other comments.  Resolving this one.